src/pykiso/lib/auxiliaries/can_auxiliary.py

```diff
@@ -0,0 +1,296 @@
+##########################################################################
+# Copyright (c) 2010-2023 Robert Bosch GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+Can Auxiliary
+*************
+
+:module: can_auxiliary
+
+:synopsis: Auxiliary enables you to read write can messages which are defined in a dbc file.
+
+The goal of this module is to be able to receive and send signals which are defined in dbc file.
+
+.. currentmodule:: can_auxiliary
+"""
+
+import logging
+import queue
+import time
+import threading
+import functools
+from typing import Any, Optional
+
+from pykiso.connector import CChannel
+from pykiso.lib.auxiliaries.communication_auxiliary import CommunicationAuxiliary
+from pykiso.interfaces.dt_auxiliary import (DTAuxiliaryInterface, open_connector, close_connector)
+from pykiso import Message
+from pykiso.can_message import CanMessage
+from pykiso.can_parser import CanMessageParser
+from pykiso.exceptions import AuxiliaryNotStarted
+
+from contextlib import ContextDecorator
+
+log = logging.getLogger(__name__)
+
+
+class _collect_messages(ContextDecorator):
+    """Context manager and decorator for the communication auxiliary
+    allowing messages collection (putting them in the queue).
+    """
+
+    def __init__(self, com_aux: CommunicationAuxiliary):
+        """Constructor used to inherit some of communication auxiliary
+        features.
+        """
+        self.com_aux = com_aux
+
+    def __enter__(self):
+        """Set the queue event to allow messages collection."""
+        log.internal_debug("Start queueing received messages.")
+        self.com_aux.queueing_event.set()
+
+    def __exit__(self, *exc):
+        """Clear queue event to stop messages collection."""
+        log.internal_debug("Stop queueing received messages.")
+        self.com_aux.queueing_event.clear()
+
+
+class CanAuxiliary(DTAuxiliaryInterface):
+    """Auxiliary is used for reading and writing can messages defined in dbc file """
+
+    def __init__(self, com: CChannel, dbc_file: str, **kwargs):
+        """Constructor.
+
+        :param com: CChannel that supports raw communication over CAN
+        """
+        super().__init__(
+            is_proxy_capable=True, tx_task_on=True, rx_task_on=True, **kwargs
+        )
+        self.tx_task_on = False
+        self.recv_can_messages = {}
+        self.channel = com
+        self.queue_tx = queue.Queue()
+        self.queueing_event = threading.Event()
+        self.collect_messages = functools.partial(_collect_messages, com_aux=self)
```

Comment:
These 3 attributes don't seem to be used, you could remove the `_collect_messages` class above too

Reply:
Ok, i will remove it and test it